### ISSUE TITLE
BIOS error beeps?

### ISSUE THREAD

**Issue by bocke** [skiselev/8088_bios#48]
Hi Sergey,

Thanx for the hard work on 8088_bios. :)

I recently bought Book8088 and it gives me a beep error while booting. As their BIOS is based on yours I was looking for the information on beep codes for 8088_bios but couldn't find it.

I did found two places with beeps in the source. A long beep means something is wrong with CPU. Two beeps means something is wrong with memory.

What would one long and two short beeps mean? Is this combination of those beeps above (I hope not), or it means something completely different?

After this, it stays turned on, but nothing is happening. The screen is empty and shows only "No video signal", so there is no output.

**Comment by bocke** [comment]
On further examination, I think the CPU is likely faulty. I'll order the replacement.

Anyway, I'm still interested in the list of exising "beep" error codes for 8088_bios. :)

**Comment by ImanolBarba** [comment]
For the book8088 BIOS, the whole beeping routine seems to be different to this one.

AFAICT from the book8088 BIOS source code (which as we all know is ripped from this one), it beeps at what is _supposed_ to be 200 Hz according to the source code (feels higher in pitch), once during the 32KB min memory test and then twice if all tests passed.

The timing of the beep function has also been butchered and it seems like it's doubling the delay logic for whatever reason.

In short, on a healthy machine (mine seems to work okay), it goes

BEEP (small pause) BEEP BEEP

I also replicated this on pcem

So it would appear maybe the screen or something in the CGA circuitry is wrong IMO

**Comment by bocke** [comment]
Oh, thank you. That's very  helpful. 

Sorry for bothering you with their modifications.  I tought they just took your code verbatim and just changed the copyright notice. :)
